tx.go

```diff
@@ -82,57 +81,21 @@ func NewTxFromBytes(b []byte) (*Tx, error) {
 // Despite the name, this is not actually reading a stream in the true sense: it is a byte slice that contains
 // many transactions one after another.
 func NewTxFromStream(b []byte) (*Tx, int, error) {
-	if len(b) < 10 {
-		return nil, 0, ErrTxTooShort
-	}
-
-	var offset int
-	t := Tx{
-		Version: binary.LittleEndian.Uint32(b[offset:4]),
-	}
-	offset += 4
+	tx := Tx{}
 
-	inputCount, size := NewVarIntFromBytes(b[offset:])
-	offset += size
+	bytesRead, err := tx.ReadFrom(bytes.NewReader(b))
 
-	// create Inputs
-	var i uint64
-	var err error
-	var input *Input
-	for ; i < uint64(inputCount); i++ {
-		input, size, err = newInputFromBytes(b[offset:])
-		if err != nil {
-			return nil, 0, err
-		}
-		offset += size
-		t.addInput(input)
-	}
-
-	// create Outputs
-	var outputCount VarInt
-	var output *Output
-	outputCount, size = NewVarIntFromBytes(b[offset:])
-	offset += size
-	for i = 0; i < uint64(outputCount); i++ {
-		output, size, err = newOutputFromBytes(b[offset:])
-		if err != nil {
-			return nil, 0, err
-		}
-		offset += size
-		t.AddOutput(output)
-	}
-
-	t.LockTime = binary.LittleEndian.Uint32(b[offset:])
-	offset += 4
-
-	return &t, offset, nil
+	return &tx, int(bytesRead), err
 }
 
 // ReadFrom reads from the `io.Reader` into the `bt.Tx`.
 func (tx *Tx) ReadFrom(r io.Reader) (int64, error) {
 	*tx = Tx{}
 	var bytesRead int64
 
+	var n64 int64
```

Comment:
Is there a reason these are defined here when they could be defined below such as `n64, err := inputCount.ReadFrom(r)`

Reply:
This was simply to stop the linter complaining about shadowed variables.  I have added a comment.